Item list things and forge!!!!!!

viciscat opened the pull request:
- Make recipes mod independent
- Forge yipee
- Did not do JEI cuz what in the hell is hapenning in there

There's probably some cleaning to do but i'm so done with this gaaahhh

viciscat committed:
item list shenanigans

viciscat committed:
emi, finish forge and recipe book

viciscat committed:
remove emi runtime

kevinthegreat1 commented on src/main/resources/skyblocker.mixins.json:
???

kevinthegreat1 committed:
Refactor recipe

kevinthegreat1 committed:
Render forge recipe duration

kevinthegreat1 reviewed (changes requested):
I added forge duration with 203d4a2. You might want to look at that. Also what's going on with `CreeperEntityAccessor`? Other than that looks good.

viciscat committed:
Multiple things
Remove unused accessor
Remove REI runtime
Make EMI and recipe book call render method of a recipe

kevinthegreat1 reviewed (dismissed):
Not tested.

AzureAaron commented:
Removed merge me label because half of this just won't work with 1.21.2.

viciscat committed:
work with 1.21.3

viciscat committed:
REI priority + config

Emirlol commented:
Why do REI and recipe book use furnace to represent forge category while EMI uses a lava bucket?
### Recipe book:
https://github.com/viciscat/Skyblocker_vic/blob/2fe34c196827827e82bff1eefab65114b6e7489d/src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipebook/SkyblockRecipeResults.java#L250
### REI:
https://github.com/viciscat/Skyblocker_vic/blob/2fe34c196827827e82bff1eefab65114b6e7489d/src/main/java/de/hysky/skyblocker/compatibility/rei/SkyblockerREIClientPlugin.java#L29
### EMI:
https://github.com/viciscat/Skyblocker_vic/blob/2fe34c196827827e82bff1eefab65114b6e7489d/src/main/java/de/hysky/skyblocker/compatibility/emi/SkyblockerEMIPlugin.java#L26
https://github.com/viciscat/Skyblocker_vic/blob/2fe34c196827827e82bff1eefab65114b6e7489d/src/main/java/de/hysky/skyblocker/compatibility/emi/SkyblockerEMIPlugin.java#L43

Emirlol commented on src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipes/SkyblockForgeRecipe.java:
```diff
@@ -0,0 +1,111 @@
+
+    @Override
+    public Text getExtraText() {
+        return Text.empty();
```
Forge items can have requirements too, this could return the craft text like `SkyblockCraftingRecipe` does.

viciscat replied:
not in neu repo parser, opened a pr

Emirlol commented on src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipebook/SkyblockRecipeResults.java:
```diff
@@ -143,6 +152,7 @@ private void drawRecipeDisplay(DrawContext context, TextRenderer textRenderer, i
 
 		//Draw the arrow that points to the recipe's result
 		context.drawTextWithShadow(textRenderer, "▶", x + 96, y + 90, 0xaaffffff);
+		if (recipeIcon != null) context.drawItem(recipeIcon, x + 115, y + 61);
```
Forging time is not displayed in the recipe book. This could probably be added with a call to `SkyblockRecipe#render` here.

Emirlol commented on src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipes/SkyblockForgeRecipe.java:
```diff
@@ -0,0 +1,111 @@
+        // Render the duration of the recipe in hours by dividing by 3600
+        ScreenPos arrowLocation = getArrowLocation(width, height);
+        if (arrowLocation != null)
+            context.drawCenteredTextWithShadow(MinecraftClient.getInstance().textRenderer, String.format("%d h", duration / 3600), arrowLocation.x() + 12, arrowLocation.y() - 10, 0xFFFFFF);
```
This could be formatted more precisely by extracting the string logic from `SkyblockTime#formatTime` to another method that only returns the string while keeping the text returning method as well.

That method could be improved further by prepending a space before each number rather than after the time letter too, while not adding that single space if the string is already empty to not add an useless leading space.

viciscat committed:
Merge branch 'master' into item-list-things

# Conflicts:
#	src/main/java/de/hysky/skyblocker/compatibility/emi/SkyblockerEMIPlugin.java
#	src/main/java/de/hysky/skyblocker/compatibility/jei/SkyblockerJEIPlugin.java
#	src/main/java/de/hysky/skyblocker/compatibility/rei/SkyblockCraftingDisplayGenerator.java
#	src/main/java/de/hysky/skyblocker/compatibility/rei/SkyblockerREIClientPlugin.java
#	src/main/resources/assets/skyblocker/lang/en_us.json

viciscat committed:
requested changes and uh stuff
ye

viciscat committed:
Merge branch 'master' into item-list-things

# Conflicts:
#	src/main/java/de/hysky/skyblocker/skyblock/itemlist/ItemRepository.java
#	src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipebook/SkyblockRecipeResults.java
#	src/main/resources/assets/skyblocker/lang/ru_ru.json
#	src/main/resources/assets/skyblocker/lang/zh_cn.json

viciscat committed:
Merge branch 'master' into item-list-things

# Conflicts:
#	build.gradle
#	src/main/java/de/hysky/skyblocker/compatibility/rei/SkyblockerREIClientPlugin.java
#	src/main/java/de/hysky/skyblocker/skyblock/itemlist/ItemRepository.java
#	src/main/resources/assets/skyblocker/lang/zh_cn.json

viciscat committed:
update REI

LifeIsAParadox commented on src/main/java/de/hysky/skyblocker/compatibility/rei/SkyblockRecipeCategory.java:
```diff
@@ -0,0 +1,94 @@
+                    .backgroundEnabled(outputSlot.showBackground())
+                    .entry(EntryStacks.of(outputSlot.stack())));
+        }
+        out.add(Widgets.wrapRenderer((Rectangle) null, (context, bounds1, mouseX, mouseY, delta) -> {
```
> (Rectangle) null

could cause crash

LifeIsAParadox reviewed (commented):
This funny bug. looks like the elements doesnt get updated on resize.

https://github.com/user-attachments/assets/6a2e3fc3-81bf-4eb8-9461-b00982ee7ced

viciscat committed:
fix stuff maybe

LifeIsAParadox commented on src/main/java/de/hysky/skyblocker/compatibility/rei/SkyblockRecipeCategory.java:
```diff
@@ -0,0 +1,94 @@
+                    .backgroundEnabled(outputSlot.showBackground())
+                    .entry(EntryStacks.of(outputSlot.stack())));
+        }
+        out.add(Widgets.wrapRenderer(bounds, (context, bounds1, mouseX, mouseY, delta) -> {
```
right/left click actions are being blocked by `wrapRenderer`. you could fix this by using `createDrawableWidget` instead.
```suggestion
        out.add(Widgets.createDrawableWidget((context, mouseX, mouseY, delta) -> {
```

LifeIsAParadox committed:
fix wrapRenderer block clicks; lava bucket -> fred
